File: Examples/ObjC-DemoApp/DemoApp/ViewController.m
#import "ViewController.h"
#import <WombatAuth/WombatAuth-Swift.h>

@interface ViewController ()

@end

@implementation ViewController

- (IBAction)authorize {
    [WombatAuth.shared requestAuthorization];
}

- (IBAction)transfer {
    WMTransfer *transfer = [[WMTransfer alloc] initFrom: @"aaaabbbbcccc" to: @"xxxxyyyyzzzz" amount: 1 precision: 4 contract: @"eosio.token" symbol: @"EOS" memo: @"Some memo"];
    [WombatAuth.shared requestTransfer: transfer];
}

- (IBAction)sign {
    [WombatAuth.shared requestSignatureWithAccount: @"account" data: @"some data to sign"];
}

- (IBAction)push {
    WMAuth *auth = [[WMAuth alloc] initWithActor: @"aaaabbbbcccc" permission: @"active"];
    NSArray<WMAction *> *actions = @[[[WMAction alloc] initWithAccount: @"eosio.token" name: @"transfer" auth: @[auth] data: @{
        @"from": @"aaaabbbbcccc",
        @"to": @"xxxxyyyyzzzz",
        @"quantity": @"1.3000 EOS",
        @"memo": @"Here you go"
    }]];
    WMTransaction *transaction = [[WMTransaction alloc] initFrom: @"aaaabbbbcccc" actions: actions];
    [WombatAuth.shared pushTransaction: transaction];
}

@end

File: Example/DemoApp/ViewController.swift
import UIKit
import WombatAuth

class ViewController: UIViewController {
    @IBAction func authorize() {
        WombatAuth.shared.requestAuthorization()
    }

    @IBAction func transfer() {
        let transfer = WMTransfer(
            from: "aaaabbbbcccc",
            to: "xxxxyyyyzzzz",
            amount: 1,
            precision: 4,
            contract: "eosio.token",
            symbol: "EOS",
            memo: "Some memo"
        )

        WombatAuth.shared.requestTransfer(transfer)
    }

    @IBAction func sign() {
        WombatAuth.shared.requestSignature(account: "account", data: "some data to sign")
    }

    @IBAction func push() {
        let transaction = WMTransaction(
            from: "aaaabbbbcccc",
            actions: [
                .init(
                    account: "eosio.token",
                    name: "transfer",
                    auth: [.init(actor: "aaaabbbbcccc", permission: "active")],
                    data: [
                        "from": "aaaabbbbcccc",
                        "to": "xxxxyyyyzzzz",
                        "quantity": "1.3000 EOS",
                        "memo": "Here you go"
                    ]
                )
            ]
        )

        WombatAuth.shared.pushTransaction(transaction)
    }
}
